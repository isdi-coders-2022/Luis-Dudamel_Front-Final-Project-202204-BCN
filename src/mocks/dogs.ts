import { IDog } from "../interfaces/Dogs";

export const MockFavDogs: IDog[] = [
  {
    age: 7,
    name: "Rocko",
    picture: "rocko.jpg",
    title: "Ball Player",
    toy: "Ball",
    weight: "4",
    breed: "beagle",
    personality: "ball",
  },
  {
    age: 2,
    name: "Lamu",
    picture: "rocko.jpg",
    title: "Beach Lover",
    toy: "Ball",
    weight: "4",
    breed: "cocker",
    personality: "beach",
  },
  {
    age: 4,
    name: "Firulais",
    picture: "rocko.jpg",
    title: "Nature Explorer",
    toy: "Ball",
    weight: "4",
    breed: "border collie",
    personality: "nature",
  },
  {
    age: 4,
    name: "Breck",
    picture: "rocko.jpg",
    title: "Calm Walker",
    toy: "Ball",
    weight: "4",
    breed: "golden retriever",
    personality: "walker",
  },
  {
    age: 9,
    name: "Fenix",
    picture: "rocko.jpg",
    title: "Frisbee Player",
    toy: "Ball",
    weight: "4",
    breed: "collie",
    personality: "frisbee",
  },
  {
    age: 9,
    name: "Chuletas",
    picture: "rocko.jpg",
    title: "Calm Walker",
    toy: "Ball",
    weight: "4",
    breed: "beagle",
    personality: "walker",
  },
];
