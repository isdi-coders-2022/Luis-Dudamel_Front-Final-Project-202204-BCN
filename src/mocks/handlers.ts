import { rest } from "msw";

export const handlers = [
  rest.post(
    `${process.env.REACT_APP_API_URL_DEV}user/register`,
    (req, res, ctx) => {
      return res(ctx.status(201));
    }
  ),
];
