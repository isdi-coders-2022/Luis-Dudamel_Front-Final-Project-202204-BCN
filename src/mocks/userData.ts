export const mockToken: string =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiTHVpcyIsInVzZXJuYW1lIjoibHVpczEiLCJpZCI6IjYyOTY2MDM2Yjc0OTY5MjUxNDk2YmZmZCIsImlhdCI6MTY1NDE5ODQyNn0.H3MfL22YtoY3QFKSl11V7H_2DBhD6pB64O75Hllm2O8";
