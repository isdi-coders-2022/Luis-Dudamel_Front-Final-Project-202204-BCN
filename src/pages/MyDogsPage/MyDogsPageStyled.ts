import styled from "styled-components";

const MyDogsPageStyled = styled.div`
  h1 {
    padding-top: 30px;
  }
`;
export default MyDogsPageStyled;
