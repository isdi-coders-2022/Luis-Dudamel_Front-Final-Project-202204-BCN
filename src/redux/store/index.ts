import { configureStore } from "@reduxjs/toolkit";
import uiSliceReducer from "../feature/usersSlice/uiSlice";
import userSliceReducer from "../feature/usersSlice/usersSlice";

const store = configureStore({
  reducer: {
    user: userSliceReducer,
    ui: uiSliceReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
