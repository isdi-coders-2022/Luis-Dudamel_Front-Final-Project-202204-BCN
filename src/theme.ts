import { createTheme } from "@mui/material/styles";
import { red } from "@mui/material/colors";

const theme = createTheme({
  typography: {
    fontFamily: ["Outfit"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "none",
          "&:hover": {
            backgroundColor: "#2a9d8f",
          },
        },
      },
    },
  },
  palette: {
    background: {
      default: "#ffc027",
    },
    primary: {
      main: "#264653",
    },
    secondary: {
      main: "#19857b",
    },
    error: {
      main: red.A400,
    },
  },
});

export default theme;
