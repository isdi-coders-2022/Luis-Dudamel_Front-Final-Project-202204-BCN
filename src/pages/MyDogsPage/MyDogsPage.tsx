import { useAppDispatch, useAppSelector } from "../../redux/hooks";
import MyDogsPageStyled from "./MyDogsPageStyled";
import { IDog } from "../../interfaces/Dogs";
import { getFavDogsThunk } from "../../redux/thunks/dogsThunks";
import { useEffect } from "react";
import { UserState } from "../../interfaces/UserCredential";
import { DogList } from "../../components/DogList/DogList";
import { Stack } from "@mui/material";

const MyDogsPage = (): JSX.Element => {
  const currentFavDogs: IDog[] = useAppSelector((state) => state.dogs);
  const currentUser: UserState = useAppSelector((state) => state.user);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(getFavDogsThunk(currentUser.username));
  }, [currentUser.username, dispatch]);

  return (
    <>
      <MyDogsPageStyled>
        <Stack
          direction="column"
          justifyContent="center"
          alignItems="center"
          spacing={1}
        >
          <h1>My dogs</h1>
          <DogList dogs={currentFavDogs}></DogList>
        </Stack>
      </MyDogsPageStyled>
    </>
  );
};

export default MyDogsPage;
