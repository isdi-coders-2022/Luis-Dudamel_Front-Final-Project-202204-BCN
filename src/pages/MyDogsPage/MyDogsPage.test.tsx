import { ThemeProvider } from "@mui/material";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { mockFavDogs } from "../../mocks/dogs";
import store from "../../redux/store";
import theme from "../../theme";
import MyDogsPage from "./MyDogsPage";

const mockDispatch = jest.fn();

jest.mock("../../redux/hooks", () => ({
  ...jest.requireActual("../../redux/hooks"),
  useAppSelector: () => mockFavDogs,
  useAppDispatch: () => mockDispatch,
}));

describe("Given a MyDogsPage component page", () => {
  describe("When its invoked", () => {
    test("Then it should call the dispatch function with an username'", () => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <ThemeProvider theme={theme}>
              <MyDogsPage />
            </ThemeProvider>
          </BrowserRouter>
        </Provider>
      );

      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
