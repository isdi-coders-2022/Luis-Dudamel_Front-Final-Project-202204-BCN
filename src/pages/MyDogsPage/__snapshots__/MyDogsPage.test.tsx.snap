// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MyDogsPage component page When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ efjVMu"
>
  <div
    className="css-1npihzk-MuiStack-root"
  >
    <nav
      className="sc-dkzDqf jJSOkW"
    >
      <div
        className="header-container MuiBox-root css-i9gxme"
      >
        <header
          className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic appbar css-18ditn3-MuiPaper-root-MuiAppBar-root"
        >
          <div
            className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              className="MuiTypography-root MuiTypography-h6 page-title css-poizw5-MuiTypography-root"
            >
              My Dogs
            </div>
          </div>
        </header>
      </div>
    </nav>
    <nav
      className="sc-hKMtZM hhRTHw"
    >
      <div
        className="profile-bar-container MuiBox-root css-i9gxme"
      >
        <header
          className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic appbar css-18ditn3-MuiPaper-root-MuiAppBar-root"
        >
          <div
            className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
          >
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root profile-button css-9ntkjh-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Logout
              <span
                className="MuiButton-endIcon MuiButton-iconSizeMedium css-1gnd1fd-MuiButton-endIcon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LogoutRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 5h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h6c.55 0 1-.45 1-1s-.45-1-1-1H5V5z"
                  />
                  <path
                    d="m20.65 11.65-2.79-2.79c-.32-.32-.86-.1-.86.35V11h-7c-.55 0-1 .45-1 1s.45 1 1 1h7v1.79c0 .45.54.67.85.35l2.79-2.79c.2-.19.2-.51.01-.7z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root profile-button css-9ntkjh-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              New dog
              <span
                className="MuiButton-endIcon MuiButton-iconSizeMedium css-1gnd1fd-MuiButton-endIcon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </header>
      </div>
    </nav>
    <div
      className="MuiGrid-root MuiGrid-container css-1hbmzt3-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 dogs-grid css-1o156uu-MuiGrid-root"
      >
        <div
          className="sc-gsnTZi dCwVrA"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root dog-card css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
            >
              <div
                className="dog-card-top"
              >
                <div
                  className="image-container"
                >
                  <img
                    alt="Rocko avatar"
                    className="dog-card-top__avatar"
                    crossOrigin=""
                    src="uploads/images/rocko.jpg"
                    width="100%"
                  />
                </div>
                <img
                  alt="Rocko avatar"
                  className="dog-card-top__personality"
                  src="../../images/icons/mobile/personalities/inactive/ball-inactive.png"
                  width={20}
                />
              </div>
              <div
                className="dog-card-bottom"
              >
                <div>
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom dog-card-bottom__name css-hwgg1z-MuiTypography-root"
                  >
                    Rocko
                  </div>
                  <p
                    className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
                  >
                    Ball Player
                  </p>
                </div>
                <img
                  alt="Red trash can icon"
                  className="dog-card-bottom__delete"
                  onClick={[Function]}
                  src="../../images/icons/mobile/trash.png"
                  width={20}
                />
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 dogs-grid css-1o156uu-MuiGrid-root"
      >
        <div
          className="sc-gsnTZi dCwVrA"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root dog-card css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
            >
              <div
                className="dog-card-top"
              >
                <div
                  className="image-container"
                >
                  <img
                    alt="Lamu avatar"
                    className="dog-card-top__avatar"
                    crossOrigin=""
                    src="uploads/images/rocko.jpg"
                    width="100%"
                  />
                </div>
                <img
                  alt="Lamu avatar"
                  className="dog-card-top__personality"
                  src="../../images/icons/mobile/personalities/inactive/beach-inactive.png"
                  width={20}
                />
              </div>
              <div
                className="dog-card-bottom"
              >
                <div>
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom dog-card-bottom__name css-hwgg1z-MuiTypography-root"
                  >
                    Lamu
                  </div>
                  <p
                    className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
                  >
                    Beach Lover
                  </p>
                </div>
                <img
                  alt="Red trash can icon"
                  className="dog-card-bottom__delete"
                  onClick={[Function]}
                  src="../../images/icons/mobile/trash.png"
                  width={20}
                />
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 dogs-grid css-1o156uu-MuiGrid-root"
      >
        <div
          className="sc-gsnTZi dCwVrA"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root dog-card css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
            >
              <div
                className="dog-card-top"
              >
                <div
                  className="image-container"
                >
                  <img
                    alt="Firulais avatar"
                    className="dog-card-top__avatar"
                    crossOrigin=""
                    src="uploads/images/rocko.jpg"
                    width="100%"
                  />
                </div>
                <img
                  alt="Firulais avatar"
                  className="dog-card-top__personality"
                  src="../../images/icons/mobile/personalities/inactive/nature-inactive.png"
                  width={20}
                />
              </div>
              <div
                className="dog-card-bottom"
              >
                <div>
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom dog-card-bottom__name css-hwgg1z-MuiTypography-root"
                  >
                    Firulais
                  </div>
                  <p
                    className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
                  >
                    Nature Explorer
                  </p>
                </div>
                <img
                  alt="Red trash can icon"
                  className="dog-card-bottom__delete"
                  onClick={[Function]}
                  src="../../images/icons/mobile/trash.png"
                  width={20}
                />
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 dogs-grid css-1o156uu-MuiGrid-root"
      >
        <div
          className="sc-gsnTZi dCwVrA"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root dog-card css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
            >
              <div
                className="dog-card-top"
              >
                <div
                  className="image-container"
                >
                  <img
                    alt="Breck avatar"
                    className="dog-card-top__avatar"
                    crossOrigin=""
                    src="uploads/images/rocko.jpg"
                    width="100%"
                  />
                </div>
                <img
                  alt="Breck avatar"
                  className="dog-card-top__personality"
                  src="../../images/icons/mobile/personalities/inactive/walker-inactive.png"
                  width={20}
                />
              </div>
              <div
                className="dog-card-bottom"
              >
                <div>
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom dog-card-bottom__name css-hwgg1z-MuiTypography-root"
                  >
                    Breck
                  </div>
                  <p
                    className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
                  >
                    Calm Walker
                  </p>
                </div>
                <img
                  alt="Red trash can icon"
                  className="dog-card-bottom__delete"
                  onClick={[Function]}
                  src="../../images/icons/mobile/trash.png"
                  width={20}
                />
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 dogs-grid css-1o156uu-MuiGrid-root"
      >
        <div
          className="sc-gsnTZi dCwVrA"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root dog-card css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
            >
              <div
                className="dog-card-top"
              >
                <div
                  className="image-container"
                >
                  <img
                    alt="Fenix avatar"
                    className="dog-card-top__avatar"
                    crossOrigin=""
                    src="uploads/images/rocko.jpg"
                    width="100%"
                  />
                </div>
                <img
                  alt="Fenix avatar"
                  className="dog-card-top__personality"
                  src="../../images/icons/mobile/personalities/inactive/frisbee-inactive.png"
                  width={20}
                />
              </div>
              <div
                className="dog-card-bottom"
              >
                <div>
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom dog-card-bottom__name css-hwgg1z-MuiTypography-root"
                  >
                    Fenix
                  </div>
                  <p
                    className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
                  >
                    Frisbee Player
                  </p>
                </div>
                <img
                  alt="Red trash can icon"
                  className="dog-card-bottom__delete"
                  onClick={[Function]}
                  src="../../images/icons/mobile/trash.png"
                  width={20}
                />
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 dogs-grid css-1o156uu-MuiGrid-root"
      >
        <div
          className="sc-gsnTZi dCwVrA"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root dog-card css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root dog-card-content css-46bh2p-MuiCardContent-root"
            >
              <div
                className="dog-card-top"
              >
                <div
                  className="image-container"
                >
                  <img
                    alt="Chuletas avatar"
                    className="dog-card-top__avatar"
                    crossOrigin=""
                    src="uploads/images/rocko.jpg"
                    width="100%"
                  />
                </div>
                <img
                  alt="Chuletas avatar"
                  className="dog-card-top__personality"
                  src="../../images/icons/mobile/personalities/inactive/walker-inactive.png"
                  width={20}
                />
              </div>
              <div
                className="dog-card-bottom"
              >
                <div>
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom dog-card-bottom__name css-hwgg1z-MuiTypography-root"
                  >
                    Chuletas
                  </div>
                  <p
                    className="MuiTypography-root MuiTypography-body1 dog-card-bottom__title css-1cz1j3c-MuiTypography-root"
                  >
                    Calm Walker
                  </p>
                </div>
                <img
                  alt="Red trash can icon"
                  className="dog-card-bottom__delete"
                  onClick={[Function]}
                  src="../../images/icons/mobile/trash.png"
                  width={20}
                />
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
